### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by itsTony4dev**

### CODE REVIEW

File: Data Model For Job Entity.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "compensation" VARCHAR,
+  "application_link" VARCHAR,
+  "is_external" BOOLEAN DEFAULT false,
+  "company_id" INTEGER,
+  "created_at" TIMESTAMP DEFAULT (now()),
+  "updated_at" TIMESTAMP DEFAULT (now())
+);
+
+CREATE TABLE "Job_Category" (
+  "category_id" INTEGER PRIMARY KEY,
```
Comment (Hkataya) [inline Data Model For Job Entity.sql:17]: i would keep the naming simple here. We already know the table name is Job_Category, we can name the columns id and title

File: Data Model For Job Entity.sql [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+CREATE TABLE "Job_Level" (
+  "level_id" INTEGER PRIMARY KEY,
+  "level_title" VARCHAR NOT NULL
+);
+
+CREATE TABLE "Job_Views" (
+  "view_id" INTEGER PRIMARY KEY,
+  "job_id" INTEGER,
+  "total_views" INTEGER,
```
Comment (Hkataya) [inline Data Model For Job Entity.sql:29]: why is total_views added here?

**Review by Hkataya — CHANGES_REQUESTED** [review]

File: Data Model For Job Entity.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
+CREATE TABLE "Job" (
+  "id" INTEGER PRIMARY KEY,
+  "title" VARCHAR NOT NULL,
+  "level_id" INTEGER NOT NULL,
+  "category_id" INTEGER NOT NULL,
+  "location" VARCHAR,
```
[the remark this answers is not in the document]
Comment (Sami-Sh99) [reply Data Model For Job Entity.sql:6]: - [ ] give all VARCHARs a limit

**Review by Sami-Sh99 — CHANGES_REQUESTED** [review]

File: jobAPI_swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,378 @@
+openapi: 3.0.3
+info:
+  title: Job Board API - OpenAPI 3.0
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:3]: Remove ` - OpenAPI 3.0`

File: jobAPI_swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+  description: This is a Job Board API based on the OpenAPI 3.0 specification.
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:4]: `Job Board RESTful API based on the OpenAPI 3.0 specification. It includes CRUD methods and filtering capabilities for managing job postings.`

File: jobAPI_swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+  version: 1.0.1
+  contact:
+    email: apiteam@jobboard.com
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:7]: remove email

File: jobAPI_swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+servers:
+  - url: https://jobboard.api.com/v1
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:12]: remove servers

File: jobAPI_swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+tags:
+  - name: job
+    description: Everything about jobs
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:15]: `User Priviledged Operations`

File: jobAPI_swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+  - name: admin
+    description: Admin operations
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:17]: `Admin privileged Operations`

File: jobAPI_swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,378 @@
...
+paths:
+  /job/{jobId}:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:19]: `/jobs/{jobId}`

File: jobAPI_swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,378 @@
...
+    get:
+      tags:
+        - job
+      summary: Get job by ID
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:23]: let's stay consistent, either `ID` or `Id`

I suggest `Id`

File: jobAPI_swagger.yaml [hunk 12]
```diff
@@ -0,0 +1,378 @@
...
+      description: Returns a single job
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:24]: `Returns a job of the given Id`

File: jobAPI_swagger.yaml [hunk 13]
```diff
@@ -0,0 +1,378 @@
...
+        - name: jobId
+          in: path
+          description: ID of job to return
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Successful operation
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:36]: just `success` all lowercase

File: jobAPI_swagger.yaml [hunk 14]
```diff
@@ -0,0 +1,378 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Job'          
+        '400':
+          description: Invalid ID supplied
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:42]: `invalid Id supplied`

File: jobAPI_swagger.yaml [hunk 15]
```diff
@@ -0,0 +1,378 @@
...
+        '404':
+          description: Job not found
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:44]: lowercase

File: jobAPI_swagger.yaml [hunk 16]
```diff
@@ -0,0 +1,378 @@
...
+        '422':
+          description: Validation exception
+  /jobs:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:98]: This should be the first element in the list (before /job/{jobId})

File: jobAPI_swagger.yaml (hunk 10 above)
Comment (Sami-Sh99) [reply jobAPI_swagger.yaml:19]: path should start in plural `jobs` fix it for the whole file

File: jobAPI_swagger.yaml [hunk 17]
```diff
@@ -0,0 +1,378 @@
...
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '204':
+          description: Job deleted successfully
+        '404':
+          description: Job not found
+    put:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:64]: add a `PATCH` method too

File: jobAPI_swagger.yaml [hunk 18]
```diff
@@ -0,0 +1,378 @@
...
+              schema:
+                $ref: '#/components/schemas/Job'          
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Job not found
+    delete:
+      tags:
+        - admin
+      summary: Delete a particular job
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:48]: ... by Id

File: jobAPI_swagger.yaml [hunk 19]
```diff
@@ -0,0 +1,378 @@
...
+          description: Invalid ID supplied
+        '404':
+          description: Job not found
+        '422':
+          description: Validation exception
+  /jobs:
+    get:
+      tags:
+        - job
+      summary: Get jobs with filters
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:102]: `... with optional filters`

File: jobAPI_swagger.yaml [hunk 20]
```diff
@@ -0,0 +1,378 @@
...
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Job'
+        '404':
+          description: No jobs found for the given filters
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:138]: or there are no jobs? filters are not always provided

File: jobAPI_swagger.yaml [hunk 21]
```diff
@@ -0,0 +1,378 @@
...
+  /job/submit:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:139]: drop the /submit
POST /jobs already refers that a job is submitted for creation

File: jobAPI_swagger.yaml [hunk 22]
```diff
@@ -0,0 +1,378 @@
...
+      requestBody:
+        description: Job details to be submitted for approval
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Job'
+      responses:
+        '202':
+          description: Job submitted for approval
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:155]: `job submitted successfully, pending approval`

File: jobAPI_swagger.yaml [hunk 23]
```diff
@@ -0,0 +1,378 @@
...
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Job published successfully
+        '404':
+          description: Job not found
+  /job/{jobId}/approve:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:180]: drop this endpoint
Admin will just send a POST /jobs/id with the approval state

File: jobAPI_swagger.yaml [hunk 24]
```diff
@@ -0,0 +1,378 @@
...
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Job approved successfully
+        '404':
+          description: Job not found
+  /job/{jobId}/reject:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:199]: drop this too

File: jobAPI_swagger.yaml [hunk 25]
```diff
@@ -0,0 +1,378 @@
...
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Job rejected successfully
+        '404':
+          description: Job not found
+  /jobcategories:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:218]: Missing other crud operations (GET, PUT, DELETE)

File: jobAPI_swagger.yaml [hunk 26]
```diff
@@ -0,0 +1,378 @@
...
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/JobCategory'
+      responses:
+        '201':
+          description: Job category created successfully
+        '400':
+          description: Invalid input
+  /joblevels:
```
Comment (Sami-Sh99) [inline jobAPI_swagger.yaml:236]: Missing other crud operations (GET, PUT, DELETE)

**Review by Sami-Sh99 — CHANGES_REQUESTED** [review]

**Pull request merged**